src/lib/alarmMgr/AlarmManager.cpp

```diff
@@ -589,6 +608,74 @@ bool AlarmManager::forwardingErrorReset(const std::string& url)
 
 
 
+/* ****************************************************************************
+*
+* AlarmManager::notificationQueue -
+*
+* Returns false if no effective alarm transition occurs, otherwise, true is returned.
+*/
+bool AlarmManager::notificationQueue(const std::string& service, const std::string& details)
+{
+  semTake();
+
+  std::map<std::string, int>::iterator iter = notificationQ.find(details);
+
+  if (iter != notificationQ.end())
+  {
+    iter->second += 1;
+
+    if (notificationQueueLogAlways)
+    {
+      LM_W(("Repeated NotificationQueue %s: %s", service.c_str(), details.c_str()));
+    }
+    else
+    {
+      // even if repeat alarms is off, this is a relevant event in debug level
+      LM_T(LmtCPrForwardRequestPayload, ("Repeated NotificationQueue %s: %s", service.c_str(), details.c_str()));
+    }
+
+    semGive();
+    return false;
+  }
+
+  ++notificationQueues;
+
+  notificationQ[details] = 1;
```

Comment:
You should use `service` as key for this map.

```suggestion
  notificationQ[service] = 1;
```

Reply:
Fixed